src/constants/index.tsx

```diff
@@ -304,61 +302,113 @@ export const PINNED_PAIRS: {
     [DAI[ChainId.MAINNET], USDT[ChainId.MAINNET]],
   ],
 }
-
+export const MAINNET_PROVIDER = new providers.JsonRpcProvider(`https://mainnet.infura.io/v3/${INFURA_PROJECT_ID}`)
 export const ARBITRUM_ONE_PROVIDER = new providers.JsonRpcProvider('https://arb1.arbitrum.io/rpc')
 
+export enum ConnectorType {
+  METAMASK = 'METAMASK',
+  COINBASE = 'COINBASE',
+  WALLET_CONNECT = 'WALLET_CONNECT',
+  NETWORK = 'NETWORK',
+}
+
 export interface WalletInfo {
-  connector?: AbstractConnector
   name: string
-  iconName: string
+  logo: string
   description: string
-  href: string | null
+  link: string | null
   color: string
-  primary?: true
   mobile?: true
   mobileOnly?: true
 }
 
 export const SUPPORTED_WALLETS: { [key: string]: WalletInfo } = {
-  INJECTED: {
-    connector: injected,
-    name: 'Injected',
-    iconName: RightArrow,
-    description: 'Injected web3 provider.',
-    href: null,
-    color: '#010101',
-    primary: true,
-  },
-  METAMASK: {
-    connector: injected,
+  [ConnectorType.METAMASK]: {
     name: 'MetaMask',
-    iconName: Metamask,
+    logo: MetaMaskLogo,
     description: 'Easy-to-use browser extension.',
-    href: null,
+    link: 'https://metamask.io/',
     color: '#E8831D',
     mobile: true,
   },
-  WALLET_CONNECT: {
-    connector: walletConnect,
+  [ConnectorType.COINBASE]: {
+    name: 'Coinbase Wallet',
+    logo: CoinbaseWalletLogo,
+    description: 'Connect using Coinbase Wallet.',
+    link: null,
+    color: '#4196FC',
+    mobile: true,
+  },
+  [ConnectorType.WALLET_CONNECT]: {
     name: 'WalletConnect',
-    iconName: WalletConnect,
+    logo: WalletConnectLogo,
     description: 'Connect to Trust Wallet, Rainbow Wallet and more...',
-    href: null,
+    link: null,
     color: '#4196FC',
     mobile: true,
   },
-  COINBASE: {
-    connector: walletLink,
-    name: 'Coinbase Wallet',
-    iconName: Coinbase,
-    description: 'Connect using Coinbase Wallet.',
-    href: null,
-    color: '#4196FC',
+  [ConnectorType.NETWORK]: {
+    name: 'Network',
+    logo: '',
+    description: 'Web3 Network',
+    link: null,
+    color: '#000000',
     mobile: true,
   },
 }
 
-export const NetworkContextName = 'NETWORK'
+export const BACKFILLABLE_WALLETS = [ConnectorType.METAMASK, ConnectorType.COINBASE, ConnectorType.WALLET_CONNECT]
+
+export const SUPPORTED_NETWORKS: { [key: string]: number[] } = {
+  [ConnectorType.METAMASK]: [
+    ChainId.MAINNET,
+    ChainId.RINKEBY,
+    ChainId.ARBITRUM_ONE,
+    ChainId.ARBITRUM_RINKEBY,
+    ChainId.XDAI,
+    ChainId.POLYGON,
+    ChainId.GOERLI,
+    ChainId.ARBITRUM_GOERLI,
+    ChainId.OPTIMISM_MAINNET,
+    ChainId.OPTIMISM_GOERLI,
```

Comment:
maybe instead of doing it manually you can do

```ts
const getSupportedChainsByConnector = () => {
  const CONNECTOR_SUPPORTED_NETWORKS: { [key in ConnectorType]: ChainId[] } = {
    COINBASE: [],
    METAMASK: [],
    NETWORK: [],
    WALLET_CONNECT: [],
  }

  const chains = Object.values(ChainId).filter(v => typeof v !== 'string') as ChainId[]

  for (const connector of Object.values(ConnectorType)) {
    CONNECTOR_SUPPORTED_NETWORKS[connector] = chains
  }

  return CONNECTOR_SUPPORTED_NETWORKS
}

export const SUPPORTED_NETWORKS = getSupportedChainsByConnector()
```

Reply:
Nice one! Implemented with [71b3da9](https://github.com/levelkdev/swapr-dapp/pull/1330/commits/71b3da9413926ebc41b732f411efdea644bc1809)